File: app/src/main/java/com/callmangement/ui/distributor/activity/PosDistributionFormViewActivity.java
package com.callmangement.ui.distributor.activity;

import androidx.annotation.NonNull;
import androidx.lifecycle.ViewModelProviders;
import android.annotation.SuppressLint;
import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.ArrayAdapter;
import com.bumptech.glide.Glide;
import com.callmangement.Network.APIService;
import com.callmangement.Network.RetrofitInstance;
import com.callmangement.R;
import com.callmangement.custom.CustomActivity;
import com.callmangement.databinding.ActivityPosDistributionFormViewBinding;
import com.callmangement.model.district.ModelDistrictList;
import com.callmangement.model.pos_distribution_form.ModelEquipmentModel;
import com.callmangement.model.pos_distribution_form.ModelNewMachineMake;
import com.callmangement.model.pos_distribution_form.ModelOldMachineMake;
import com.callmangement.ui.complaint.ComplaintViewModel;
import com.callmangement.ui.distributor.model.PosDistributionDetail;
import com.callmangement.ui.distributor.model.PosDistributionFormDetailResponse;
import com.callmangement.ui.home.ZoomInZoomOutActivity;
import com.callmangement.utils.Constants;
import com.callmangement.utils.PrefManager;
import java.util.ArrayList;
import java.util.List;
import okhttp3.ResponseBody;
import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;

public class PosDistributionFormViewActivity extends CustomActivity implements View.OnClickListener {
    private ActivityPosDistributionFormViewBinding binding;
    private PrefManager prefManager;
    private PosDistributionDetail model;
    private final List<ModelEquipmentModel> listEquipmentModel = new ArrayList<>();
    private final List<ModelOldMachineMake> listOldMachineMake = new ArrayList<>();
    private final List<ModelNewMachineMake> listNewMachineMake = new ArrayList<>();
    private ComplaintViewModel viewModel;
    private List<ModelDistrictList> district_List = new ArrayList<>();
    private String photoPath = "";
    private String formPhotoPath = "";
    
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        binding = ActivityPosDistributionFormViewBinding.inflate(getLayoutInflater());
        setContentView(binding.getRoot());
        binding.actionBar.ivBack.setVisibility(View.VISIBLE);
        binding.actionBar.ivThreeDot.setVisibility(View.GONE);
        binding.actionBar.layoutLanguage.setVisibility(View.GONE);
        binding.actionBar.buttonPDF.setVisibility(View.GONE);
        binding.actionBar.textToolbarTitle.setText(getResources().getString(R.string.pos_distribution_form));
        viewModel = ViewModelProviders.of(this).get(ComplaintViewModel.class);
        prefManager = new PrefManager(mContext);
        model = (PosDistributionDetail) getIntent().getSerializableExtra("param");
        initView();
    }

    private void initView() {

        setUpClickListener();
        setEquipmentModelSpinner();
        setNewMachineMakeSpinner();
        setOldMachineMakeSpinner();
        districtList();
        getPosDistributionDetail();
    }

    private void setUpClickListener(){
        binding.actionBar.ivBack.setOnClickListener(this);
        binding.rlPhoto.setOnClickListener(this);
        binding.rlUploadFormPhoto.setOnClickListener(this);
        binding.actionBar.buttonPDF.setOnClickListener(this);
    }

    private void setEquipmentModelSpinner() {
        ModelEquipmentModel model = new ModelEquipmentModel();
        model.setId("1");
        model.setName("Mobiocean TPS 900 (Android 10 OS)");
        listEquipmentModel.add(model);
        ArrayAdapter<ModelEquipmentModel> dataAdapter = new ArrayAdapter<>(mContext, android.R.layout.simple_spinner_item, listEquipmentModel);
        dataAdapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item);
        binding.spinnerEquipmentModel.setAdapter(dataAdapter);
    }

    private void setOldMachineMakeSpinner() {
        ModelOldMachineMake model1 = new ModelOldMachineMake();
        model1.setId("1");
        model1.setName("VISIONTEK");
        listOldMachineMake.add(model1);
        ModelOldMachineMake model2 = new ModelOldMachineMake();
        model2.setId("2");
        model2.setName("ANALOGICS");
        listOldMachineMake.add(model2);
        ArrayAdapter<ModelOldMachineMake> dataAdapter = new ArrayAdapter<>(mContext, android.R.layout.simple_spinner_item, listOldMachineMake);
        dataAdapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item);
        binding.spinnerOldMachineMake.setAdapter(dataAdapter);
    }

    private void setNewMachineMakeSpinner() {
        ModelNewMachineMake model1 = new ModelNewMachineMake();
        model1.setId("3");
        model1.setName("MOBIOCEAN");
        listNewMachineMake.add(model1);
        ArrayAdapter<ModelNewMachineMake> dataAdapter = new ArrayAdapter<>(mContext, android.R.layout.simple_spinner_item, listNewMachineMake);
        dataAdapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item);
        binding.spinnerNewMachineMake.setAdapter(dataAdapter);
    }

    private void districtList() {
        if (Constants.isNetworkAvailable(mContext)) {
            isLoading();
            viewModel.getDistrict().observe(this, modelDistrict -> {
                isLoading();
                if (modelDistrict.status.equals("200")) {
                    district_List = modelDistrict.district_List;
                    if (district_List != null && district_List.size() > 0) {
                        ArrayAdapter<ModelDistrictList> dataAdapter = new ArrayAdapter<>(mContext, android.R.layout.simple_spinner_item, district_List);
                        dataAdapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item);
                        binding.spinnerDistrict.setAdapter(dataAdapter);
                    }

                }
            });
        } else {
            makeToast(getResources().getString(R.string.no_internet_connection));
        }
    }

    private void isLoading() {
        viewModel.getIsLoading().observe(this, aBoolean -> {
            if (aBoolean) {
                showProgress(getResources().getString(R.string.please_wait));
            } else {
                hideProgress();
            }
        });
    }

    private void getPosDistributionDetail(){
        if (Constants.isNetworkAvailable(mContext)) {
            showProgress();
            APIService service = RetrofitInstance.getRetrofitInstance().create(APIService.class);
            Call<ResponseBody> call = service.getPosDistributionDetailByTIDAPI( prefManager.getUSER_Id(), String.valueOf(model.getTranId()));
            call.enqueue(new Callback<ResponseBody>() {
                @SuppressLint("SetTextI18n")
                @Override
                public void onResponse(@NonNull Call<ResponseBody> call, @NonNull Response<ResponseBody> response) {
                    hideProgress();
                    if (response.isSuccessful()) {
                        try {
                            if (response.code() == 200) {
                                if (response.body() != null) {
                                    String responseStr = response.body().string();
                                    PosDistributionFormDetailResponse modelResponse = (PosDistributionFormDetailResponse) getObject(responseStr, PosDistributionFormDetailResponse.class);
                                    PosDistributionDetail posDistributionDetail = modelResponse.getPosDistributionDetail();
                                    if (modelResponse.getStatus().equals("200")) {
                                        if (posDistributionDetail != null){
                                            binding.inputDate.setText(posDistributionDetail.getTranDateStr());
                                            binding.inputFpsCode.setText(posDistributionDetail.getFpscode());
                                            binding.inputDealerName.setText(posDistributionDetail.getDealerName());
                                            binding.inputMobileNumber.setText(posDistributionDetail.getMobileNo());
                                            binding.inputTicketNumber.setText(posDistributionDetail.getTicketNo());
                                            binding.inputBlockName.setText(posDistributionDetail.getBlockName());
                                            binding.inputOldMachineSrNo.setText(posDistributionDetail.getOldMachineSerialNo());
                                            binding.inputFingerprintSrNo.setText(posDistributionDetail.getOldMachineBiometricSeriallNo());
                                            binding.inputNewMachineOrderNo.setText(String.valueOf(posDistributionDetail.getNewMachineOrderNo()));
                                            binding.inputNewMachineSrNo.setText(posDistributionDetail.getNewMachineSerialNo());
                                            binding.inputNewFingerprintSrNo.setText(posDistributionDetail.getNewMachineBiometricSeriallNo());
                                            binding.inputIMEIIMEI2.setText(posDistributionDetail.getNewMachineIMEI1()+" - "+posDistributionDetail.getNewMachineIMEI2());
                                            binding.spinnerDistrict.setEnabled(false);
                                            binding.spinnerEquipmentModel.setEnabled(false);
                                            binding.spinnerOldMachineMake.setEnabled(false);
                                            binding.spinnerNewMachineMake.setEnabled(false);
                                            photoPath = Constants.API_BASE_URL+posDistributionDetail.getUpPhotoPath();
                                            formPhotoPath = Constants.API_BASE_URL+posDistributionDetail.getUpFormPath();
                                            Glide.with(mContext)
                                                    .load(Constants.API_BASE_URL+posDistributionDetail.getUpPhotoPath())
                                                    .placeholder(R.drawable.image_not_fount)
                                                    .into(binding.ivPhoto);
                                            Glide.with(mContext)
                                                    .load(Constants.API_BASE_URL+posDistributionDetail.getUpFormPath())
                                                    .placeholder(R.drawable.image_not_fount)
                                                    .into(binding.ivUploadFormPhoto);
                                            if (posDistributionDetail.getWhetherOldMachProvdedForReplacmnt()){
                                                binding.chkYes.setChecked(true);
                                                binding.chkNo.setChecked(false);
                                            } else {
                                                binding.chkNo.setChecked(true);
                                                binding.chkYes.setChecked(false);
                                            }
                                            if (posDistributionDetail.getOldMachineWorkingStatus().equalsIgnoreCase("Running")){
                                                binding.chkRunning.setChecked(true);
                                                binding.chkDead.setChecked(false);
                                            } else if (posDistributionDetail.getOldMachineWorkingStatus().equalsIgnoreCase("Dead")){
                                                binding.chkRunning.setChecked(false);
                                                binding.chkDead.setChecked(true);
                                            }
                                            binding.chkCompleteWithSatisfactorily.setChecked(posDistributionDetail.getIsCompleteWithSatisfactorily());
                                            if (district_List.size() > 0){
                                                for (int i = 0; i < district_List.size(); i++){
                                                    if (posDistributionDetail.getDistrictId() == Integer.parseInt(district_List.get(i).getDistrictId())){
                                                        binding.spinnerDistrict.setSelection(i);
                                                        binding.spinnerDistrict.setEnabled(false);
                                                    }
                                                }
                                            }

                                            if (listEquipmentModel.size() > 0){
                                                for (int i = 0; i < listEquipmentModel.size(); i++){
                                                    if (posDistributionDetail.getEquipmentModelId() == Integer.parseInt(listEquipmentModel.get(i).getId())){
                                                        binding.spinnerEquipmentModel.setSelection(i);
                                                        binding.spinnerEquipmentModel.setEnabled(false);
                                                    }
                                                }
                                            }

                                            if (listOldMachineMake.size() > 0){
                                                for (int i = 0; i < listOldMachineMake.size(); i++){
                                                    if (posDistributionDetail.getOldMachineVenderid() == Integer.parseInt(listOldMachineMake.get(i).getId())){
                                                        binding.spinnerOldMachineMake.setSelection(i);
                                                        binding.spinnerOldMachineMake.setEnabled(false);
                                                    }
                                                }
                                            }
                                            if (listNewMachineMake.size() > 0){
                                                for (int i = 0; i < listNewMachineMake.size(); i++){
                                                    if (posDistributionDetail.getNewMachineVenderid() == Integer.parseInt(listNewMachineMake.get(i).getId())){
                                                        binding.spinnerNewMachineMake.setSelection(i);
                                                        binding.spinnerNewMachineMake.setEnabled(false);
                                                    }
                                                }
                                            }
                                        }
                                    } else {
                                        makeToast(modelResponse.getMessage());
                                    }
                                } else {
                                    makeToast(getResources().getString(R.string.error));
                                }
                            } else {
                                makeToast(getResources().getString(R.string.error));
                            }
                        } catch (Exception e) {
                            e.printStackTrace();
                        }
                    } else {
                        makeToast(getResources().getString(R.string.error));
                    }
                }

                @Override
                public void onFailure(@NonNull Call<ResponseBody> call, @NonNull Throwable t) {
                    hideProgress();
                    makeToast(getResources().getString(R.string.error_message));
                }
            });
        } else {
            makeToast(getResources().getString(R.string.no_internet_connection));
        }
    }

    @Override
    public void onClick(View view) {
        int id = view.getId();
        if (id == R.id.iv_back) {
            onBackPressed();
        } else if (id == R.id.rl_photo) {
            startActivity(new Intent(mContext, ZoomInZoomOutActivity.class).putExtra("image", photoPath));
        } else if (id == R.id.rl_upload_form_photo) {
            startActivity(new Intent(mContext, ZoomInZoomOutActivity.class).putExtra("image", formPhotoPath));
        }
    }
}

File: app/src/main/java/com/callmangement/ui/distributor/activity/PosDistributionFormViewActivity.kt
package com.callmangement.ui.distributor.activity

import android.annotation.SuppressLint
import android.content.Intent
import android.os.Bundle
import android.view.View
import android.widget.ArrayAdapter
import androidx.lifecycle.ViewModelProviders
import com.bumptech.glide.Glide
import com.callmangement.network.APIService
import com.callmangement.network.RetrofitInstance
import com.callmangement.R
import com.callmangement.custom.CustomActivity
import com.callmangement.databinding.ActivityPosDistributionFormViewBinding
import com.callmangement.model.district.ModelDistrict
import com.callmangement.model.district.ModelDistrictList
import com.callmangement.model.pos_distribution_form.ModelEquipmentModel
import com.callmangement.model.pos_distribution_form.ModelNewMachineMake
import com.callmangement.model.pos_distribution_form.ModelOldMachineMake
import com.callmangement.ui.complaint.ComplaintViewModel
import com.callmangement.ui.distributor.model.PosDistributionDetail
import com.callmangement.ui.distributor.model.PosDistributionFormDetailResponse
import com.callmangement.ui.home.ZoomInZoomOutActivity
import com.callmangement.utils.Constants
import com.callmangement.utils.PrefManager
import okhttp3.ResponseBody
import retrofit2.Call
import retrofit2.Callback
import retrofit2.Response

class PosDistributionFormViewActivity : CustomActivity(), View.OnClickListener {
    private var binding: ActivityPosDistributionFormViewBinding? = null
    private var prefManager: PrefManager? = null
    private var model: PosDistributionDetail? = null
    private val listEquipmentModel: MutableList<ModelEquipmentModel> = ArrayList()
    private val listOldMachineMake: MutableList<ModelOldMachineMake> = ArrayList()
    private val listNewMachineMake: MutableList<ModelNewMachineMake> = ArrayList()
    private var viewModel: ComplaintViewModel? = null
    private var district_List: List<ModelDistrictList>? = ArrayList()
    private var photoPath = ""
    private var formPhotoPath = ""

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityPosDistributionFormViewBinding.inflate(
            layoutInflater
        )
        setContentView(binding!!.root)
        binding!!.actionBar.ivBack.visibility = View.VISIBLE
        binding!!.actionBar.ivThreeDot.visibility = View.GONE
        binding!!.actionBar.layoutLanguage.visibility = View.GONE
        binding!!.actionBar.buttonPDF.visibility = View.GONE
        binding!!.actionBar.textToolbarTitle.text =
            resources.getString(R.string.pos_distribution_form)
        viewModel = ViewModelProviders.of(this).get(
            ComplaintViewModel::class.java
        )
        prefManager = PrefManager(mContext)
        model = intent.getSerializableExtra("param") as PosDistributionDetail?
        initView()
    }

    private fun initView() {
        setUpClickListener()
        setEquipmentModelSpinner()
        setNewMachineMakeSpinner()
        setOldMachineMakeSpinner()
        districtList()
        posDistributionDetail
    }

    private fun setUpClickListener() {
        binding!!.actionBar.ivBack.setOnClickListener(this)
        binding!!.rlPhoto.setOnClickListener(this)
        binding!!.rlUploadFormPhoto.setOnClickListener(this)
        binding!!.actionBar.buttonPDF.setOnClickListener(this)
    }

    private fun setEquipmentModelSpinner() {
        val model = ModelEquipmentModel()
        model.id = "1"
        model.name = "Mobiocean TPS 900 (Android 10 OS)"
        listEquipmentModel.add(model)
        val dataAdapter =
            ArrayAdapter(mContext, android.R.layout.simple_spinner_item, listEquipmentModel)
        dataAdapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item)
        binding!!.spinnerEquipmentModel.adapter = dataAdapter
    }

    private fun setOldMachineMakeSpinner() {
        val model1 = ModelOldMachineMake()
        model1.id = "1"
        model1.name = "VISIONTEK"
        listOldMachineMake.add(model1)
        val model2 = ModelOldMachineMake()
        model2.id = "2"
        model2.name = "ANALOGICS"
        listOldMachineMake.add(model2)
        val dataAdapter =
            ArrayAdapter(mContext, android.R.layout.simple_spinner_item, listOldMachineMake)
        dataAdapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item)
        binding!!.spinnerOldMachineMake.adapter = dataAdapter
    }

    private fun setNewMachineMakeSpinner() {
        val model1 = ModelNewMachineMake()
        model1.id = "3"
        model1.name = "MOBIOCEAN"
        listNewMachineMake.add(model1)
        val dataAdapter =
            ArrayAdapter(mContext, android.R.layout.simple_spinner_item, listNewMachineMake)
        dataAdapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item)
        binding!!.spinnerNewMachineMake.adapter = dataAdapter
    }

    private fun districtList() {
        if (Constants.isNetworkAvailable(mContext)) {
            isLoading
            viewModel!!.district.observe(
                this
            ) { modelDistrict: ModelDistrict? ->
                isLoading
                if (modelDistrict!!.status == "200") {
                    district_List = modelDistrict.district_List
                    if (district_List != null && district_List!!.size > 0) {
                        val dataAdapter = ArrayAdapter(
                            mContext, android.R.layout.simple_spinner_item,
                            district_List!!
                        )
                        dataAdapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item)
                        binding!!.spinnerDistrict.adapter = dataAdapter
                    }
                }
            }
        } else {
            makeToast(resources.getString(R.string.no_internet_connection))
        }
    }

    private val isLoading: Unit
        get() {
            viewModel!!.isLoading.observe(
                this
            ) { aBoolean: Boolean ->
                if (aBoolean) {
                    showProgress(resources.getString(R.string.please_wait))
                } else {
                    hideProgress()
                }
            }
        }

    private val posDistributionDetail: Unit
        get() {
            if (Constants.isNetworkAvailable(mContext)) {
                showProgress()
                val service = RetrofitInstance.getRetrofitInstance().create(
                    APIService::class.java
                )
                val call = service.getPosDistributionDetailByTIDAPI(
                    prefManager!!.useR_Id, model!!.tranId.toString()
                )
                call.enqueue(object : Callback<ResponseBody?> {
                    @SuppressLint("SetTextI18n")
                    override fun onResponse(
                        call: Call<ResponseBody?>,
                        response: Response<ResponseBody?>
                    ) {
                        hideProgress()
                        if (response.isSuccessful) {
                            try {
                                if (response.code() == 200) {
                                    if (response.body() != null) {
                                        val responseStr = response.body()!!.string()
                                        val modelResponse = getObject(
                                            responseStr,
                                            PosDistributionFormDetailResponse::class.java
                                        ) as PosDistributionFormDetailResponse
                                        val posDistributionDetail =
                                            modelResponse.posDistributionDetail
                                        if (modelResponse.status == "200") {
                                            if (posDistributionDetail != null) {
                                                binding!!.inputDate.setText(posDistributionDetail.tranDateStr)
                                                binding!!.inputFpsCode.setText(posDistributionDetail.fpscode)
                                                binding!!.inputDealerName.setText(
                                                    posDistributionDetail.dealerName
                                                )
                                                binding!!.inputMobileNumber.setText(
                                                    posDistributionDetail.mobileNo
                                                )
                                                binding!!.inputTicketNumber.setText(
                                                    posDistributionDetail.ticketNo
                                                )
                                                binding!!.inputBlockName.setText(
                                                    posDistributionDetail.blockName
                                                )
                                                binding!!.inputOldMachineSrNo.setText(
                                                    posDistributionDetail.oldMachineSerialNo
                                                )
                                                binding!!.inputFingerprintSrNo.setText(
                                                    posDistributionDetail.oldMachineBiometricSeriallNo
                                                )
                                                binding!!.inputNewMachineOrderNo.setText(
                                                    posDistributionDetail.newMachineOrderNo.toString()
                                                )
                                                binding!!.inputNewMachineSrNo.setText(
                                                    posDistributionDetail.newMachineSerialNo
                                                )
                                                binding!!.inputNewFingerprintSrNo.setText(
                                                    posDistributionDetail.newMachineBiometricSeriallNo
                                                )
                                                binding!!.inputIMEIIMEI2.setText(
                                                    posDistributionDetail.newMachineIMEI1 + " - " + posDistributionDetail.newMachineIMEI2
                                                )
                                                binding!!.spinnerDistrict.isEnabled = false
                                                binding!!.spinnerEquipmentModel.isEnabled = false
                                                binding!!.spinnerOldMachineMake.isEnabled = false
                                                binding!!.spinnerNewMachineMake.isEnabled = false
                                                photoPath =
                                                    Constants.API_BASE_URL + posDistributionDetail.upPhotoPath
                                                formPhotoPath =
                                                    Constants.API_BASE_URL + posDistributionDetail.upFormPath
                                                Glide.with(mContext)
                                                    .load(Constants.API_BASE_URL + posDistributionDetail.upPhotoPath)
                                                    .placeholder(R.drawable.image_not_fount)
                                                    .into(binding!!.ivPhoto)
                                                Glide.with(mContext)
                                                    .load(Constants.API_BASE_URL + posDistributionDetail.upFormPath)
                                                    .placeholder(R.drawable.image_not_fount)
                                                    .into(binding!!.ivUploadFormPhoto)
                                                if (posDistributionDetail.whetherOldMachProvdedForReplacmnt!!) {
                                                    binding!!.chkYes.isChecked = true
                                                    binding!!.chkNo.isChecked = false
                                                } else {
                                                    binding!!.chkNo.isChecked = true
                                                    binding!!.chkYes.isChecked = false
                                                }
                                                if (posDistributionDetail.oldMachineWorkingStatus.equals(
                                                        "Running",
                                                        ignoreCase = true
                                                    )
                                                ) {
                                                    binding!!.chkRunning.isChecked = true
                                                    binding!!.chkDead.isChecked = false
                                                } else if (posDistributionDetail.oldMachineWorkingStatus.equals(
                                                        "Dead",
                                                        ignoreCase = true
                                                    )
                                                ) {
                                                    binding!!.chkRunning.isChecked = false
                                                    binding!!.chkDead.isChecked = true
                                                }
                                                binding!!.chkCompleteWithSatisfactorily.isChecked =
                                                    posDistributionDetail.isCompleteWithSatisfactorily!!
                                                if (district_List!!.size > 0) {
                                                    for (i in district_List!!.indices) {
                                                        if (posDistributionDetail.districtId == district_List!![i].districtId.toInt()) {
                                                            binding!!.spinnerDistrict.setSelection(i)
                                                            binding!!.spinnerDistrict.isEnabled =
                                                                false
                                                        }
                                                    }
                                                }

                                                if (listEquipmentModel.size > 0) {
                                                    for (i in listEquipmentModel.indices) {
                                                        if (posDistributionDetail.equipmentModelId == listEquipmentModel[i].id.toInt()) {
                                                            binding!!.spinnerEquipmentModel.setSelection(
                                                                i
                                                            )
                                                            binding!!.spinnerEquipmentModel.isEnabled =
                                                                false
                                                        }
                                                    }
                                                }

                                                if (listOldMachineMake.size > 0) {
                                                    for (i in listOldMachineMake.indices) {
                                                        if (posDistributionDetail.oldMachineVenderid == listOldMachineMake[i].id.toInt()) {
                                                            binding!!.spinnerOldMachineMake.setSelection(
                                                                i
                                                            )
                                                            binding!!.spinnerOldMachineMake.isEnabled =
                                                                false
                                                        }
                                                    }
                                                }
                                                if (listNewMachineMake.size > 0) {
                                                    for (i in listNewMachineMake.indices) {
                                                        if (posDistributionDetail.newMachineVenderid == listNewMachineMake[i].id.toInt()) {
                                                            binding!!.spinnerNewMachineMake.setSelection(
                                                                i
                                                            )
                                                            binding!!.spinnerNewMachineMake.isEnabled =
                                                                false
                                                        }
                                                    }
                                                }
                                            }
                                        } else {
                                            makeToast(modelResponse.message)
                                        }
                                    } else {
                                        makeToast(resources.getString(R.string.error))
                                    }
                                } else {
                                    makeToast(resources.getString(R.string.error))
                                }
                            } catch (e: Exception) {
                                e.printStackTrace()
                            }
                        } else {
                            makeToast(resources.getString(R.string.error))
                        }
                    }

                    override fun onFailure(call: Call<ResponseBody?>, t: Throwable) {
                        hideProgress()
                        makeToast(resources.getString(R.string.error_message))
                    }
                })
            } else {
                makeToast(resources.getString(R.string.no_internet_connection))
            }
        }

    override fun onClick(view: View) {
        val id = view.id
        if (id == R.id.iv_back) {
            onBackPressed()
        } else if (id == R.id.rl_photo) {
            startActivity(
                Intent(mContext, ZoomInZoomOutActivity::class.java).putExtra(
                    "image",
                    photoPath
                )
            )
        } else if (id == R.id.rl_upload_form_photo) {
            startActivity(
                Intent(mContext, ZoomInZoomOutActivity::class.java).putExtra(
                    "image",
                    formPhotoPath
                )
            )
        }
    }
}